docs/entities/renderer.md

```diff
@@ -0,0 +1,584 @@
+
+@SubscribeEvent
+public static void registerEntityRenderers(EntityRenderersEvent.RegisterRenderers event) {
+    event.registerEntityRenderer(MY_ENTITY_TYPE.get(), MyEntityRenderer::new);
+}
+```
+
+## Modifying Existing Entity Renderers
+
+In some scenarios, it is desirable to add to an existing entity renderer, e.g. for rendering additional effects on an existing entity. Most of the time, this will affect living entities, i.e., entities with a `LivingEntityRenderer`. This enables us to add [render layers][renderlayer] to an entity like so:
+
+```java
+@SubscribeEvent
+public static void addLayers(EntityRenderersEvent.AddLayers event) {
+    // Add a layer to every single entity type.
+    for (EntityType<?> entityType : event.getEntityTypes()) {
+        // Get our renderer.
+        EntityRenderer<?, ?> renderer = event.getRenderer(entityType);
+        // Null-check the renderer. You could add more checks here,
+        // e.g., instanceof checks to only target specific renderer classes.
+        if (renderer != null) {
+            // Add the layer to the renderer. Like above, construct a new MyRenderLayer.
+            // The EntityModelSet can be retrieved from the event through #getEntityModels.
+            renderer.addLayer(new MyRenderLayer(renderer, event.getEntityModels()));
+        }
+    }
+}
+```
+
+For players, a bit of special-casing is required because there can actually be multiple player renderers. These are managed separately by the event. We can interact with them like so:
+
+```java
+@SubscribeEvent
+public static void addPlayerLayers(EntityRenderersEvent.AddLayers event) {
+    // Iterate over all possible player models.
+    for (PlayerSkin.Model skin : event.getSkins()) {
+        // Get the associated PlayerRenderer.
+        if (event.getSkin(skin) instanceof PlayerRenderer playerRenderer) {
+            // Add the layer to the renderer.
+            playerRenderer.addLayer(new MyRenderLayer(playerRenderer, event.getEntityModels()));
+        }
+    }
+}
+```
+
+## Animations
+
+NeoForge adds an optional system that allows entity animations to be defined in JSON files, similar to third-party libraries such as [GeckoLib][geckolib]. Animations are defined in JSON files located at `assets/<namespace>/neoforge/animations/entity/<path>.json` (so for the [resource location][rl] `examplemod:example`, the file would be located at `assets/examplemod/neoforge/animations/entity/example.json`). The format of an animation file is as follows:
+
+```json5
+{
+    // The duration of the animation, in seconds.
+    "length": 1.5,
+    // Whether the animation should loop (true) or stop (false) when finished.
+    // Optional, defaults to false.
+    "loop": true,
+    // A list of parts to be animated, and their animation data.
+    "animations": [
+        {
+            // The name of the part to be animated.
+            // Must match the name of a part defined in your LayerDefinition (see above).
+            "bone": "head",
```

Comment:
If there are multiple matches, it will just pick the first one it finds, which will always be a depth-first search iirc.